### PR TITLE
EIRH-68: A11y fixes.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Implements feedback from https://docs.google.com/document/d/1G3M3ds77qwG_GgvO5QBBNR2wEU-oKJdFMTAh_acpQKc/edit#heading=h.j92tkf9km636
- Rebuilds the autocomplete component with a [custom createAutocomplete](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-core/createAutocomplete/)
- I had to rebuild the autocomplete component because the templates did not allow me at the markup and hooks that were needed to get to all of the fixes

# Review By (Date)
- EOW

# Criticality
- High

# Urgency
- High

# Review Tasks

1. Check out this branch
2. Review the items in https://docs.google.com/document/d/1G3M3ds77qwG_GgvO5QBBNR2wEU-oKJdFMTAh_acpQKc/edit#heading=h.j92tkf9km636
3. For each item, verify the fix has been met
4. Review code
5. Review the autocomplete component and ensure it works as expected
6. Review the autocomplete component and ensure it looks like it did
7. ~Don't worry about the pager feedback. Got the ok to skip that one.~ 
8. Pager fixes in https://github.com/SU-SWS/eirh-subtheme/pull/65

For Shea Todo
- Check in with Caryl about adding the aria-selected vs selected on the sortby widget as I couldn't even hard code the status and get it to show up. React bug???

### CODE REVIEW

File: src/eventshub/src/component/Autocomplete/Autocomplete.tsx
Comment (sherakama) [inline src/eventshub/src/component/Autocomplete/Autocomplete.tsx:1]: This component was more or less a re-write. Please proceed with scrutiny.

File: src/eventshub/src/component/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -28,152 +20,224 @@ type AutocompleteProps = Partial<AutocompleteOptions<AlgoliaVenueItem>> & {
...
-  ]);
+  const formProps = autocomplete.getFormProps({ inputElement: inputRef.current });
+  const inputProps = autocomplete.getInputProps({ inputElement: inputRef.current });
 
   return (
-    <>
-      <div className={cnb('er-mb-30 md:er-mb-36', className)} ref={containerRef} />
-      <div ref={panelRef} />
-    </>
+    <div className="aa-Autocomplete" {...autocomplete.getRootProps({})}>
```
Comment (sherakama) [inline src/eventshub/src/component/Autocomplete/Autocomplete.tsx:136]: Now we control almost all of the markup...

File: src/eventshub/src/component/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -28,152 +20,224 @@ type AutocompleteProps = Partial<AutocompleteOptions<AlgoliaVenueItem>> & {
...
-                  },
-                ],
-              });
-            },
-            templates: {
-              item({ item, components }) {
-                return <SuggestionHit hit={item} components={components} />;
+  const autocomplete = useMemo(
+    () =>
+      createAutocomplete<AlgoliaVenueItem, CombinedEvent, ReactMouseEvent, ReactKeyBoardEvent>({
```
Comment (sherakama) [inline src/eventshub/src/component/Autocomplete/Autocomplete.tsx:60]: New function to allow us to create our own 'headless' autocomplete. This also should allow us to skip any issues with the different react version requirements/compatibility with instantsearch.

File: src/eventshub/src/component/FilterChips/FilterChips.tsx [hunk 3]
```diff
@@ -28,7 +28,7 @@ export const FilterChips = () => {
       <Heading as='h3' size="base">Filter By</Heading>
       <div>
         {combined.map((key) => (
-          <button key={key} onClick={() => removeEventFilter(key)}
+          <button key={key} aria-label={`Clear ${key} filter`} onClick={() => removeEventFilter(key)}
```
Comment (sherakama) [inline src/eventshub/src/component/FilterChips/FilterChips.tsx:31]: Adds label to add descriptive text.

File: src/eventshub/src/component/Search/Search.tsx [hunk 4]
```diff
@@ -49,8 +49,8 @@ const Search = () => {
       <Index indexName={indexName} indexId={indexName}>
         <div>
           <Tabs activeTab={activeTab} onClick={changeTab} />
-          <div>
-            <Configure hitsPerPage={25} analytics={true} index={indexName} />
+          <div id={`panel-${activeTab}`} role="tabpanel" tabIndex={0} aria-labelledby={`tab-${activeTab}`}>
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:52]: Changes the panels to follow the panel MDN practice.

File: src/eventshub/src/component/Search/Search.tsx [hunk 5]
```diff
@@ -49,8 +49,8 @@ const Search = () => {
...
+            <Configure hitsPerPage={15} analytics={true} index={indexName} />
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:53]: Client request to reduce to 15 per pages. I slipped this in here.

File: src/eventshub/src/component/Search/Search.tsx [hunk 6]
```diff
@@ -65,15 +65,16 @@ const Search = () => {
...
-                <Autocomplete
-                  searchClient={algoliaClient}
-                  searchIndex={indexName}
-                  placeholder={placeholder}
-                  detachedMediaQuery='none'
-                  openOnFocus
-                  activeTab={activeTab}
-                  activeField={activeField}
-                />
+                <div className="er-mb-30 md:er-mb-36">
```
Comment (sherakama) [inline src/eventshub/src/component/Search/Search.tsx:68]: Moved this div here. It was inside the autocomplete component before and could go back in if the mobile stuff is janky.

File: src/eventshub/src/component/SortBy/SortBy.tsx [hunk 7]
```diff
@@ -41,10 +42,10 @@ export const SortBy = ({ activeTab }: SortByProps) => {
   return (
     <div className="er-flex er-direction-row er-items-center er-w-fit er-ml-auto">
       <label htmlFor={id} className="er-m-0 er-max-w-fit er-w-full er-text-16 er-h-fit">Sort by: </label>
-      <select id={id} onChange={handleChange} value={activeSort} className="er-m-0 er-py-0 er-pl-6 er-pr-36 er-max-w-fit er-w-full er-text-16">
-        <option value='relevance'>Relevance</option>
-        <option value='asc'>Name: A-Z</option>
-        <option value='desc'>Name: Z-A</option>
+      <select id={id} value={activeSort} onChange={handleChange} className="er-m-0 er-py-0 er-pl-6 er-pr-36 er-max-w-fit er-w-full er-text-16">
```
Comment (sherakama) [inline src/eventshub/src/component/SortBy/SortBy.tsx:45]: having the value set here like this is supposed to have the options `selected` in the rendered markup but for the life of me I can't figure out why it is able to set the element correctly visually, and has selected=true in the property inspector, but doesn't have the attribute on the HTML element!

File: src/eventshub/src/component/SortBy/SortBy.tsx [hunk 8]
```diff
@@ -41,10 +42,10 @@ export const SortBy = ({ activeTab }: SortByProps) => {
...
+        <option aria-selected={(activeSort === 'relevance') ? true : undefined} value='relevance'>Relevance</option>
+        <option aria-selected={(activeSort === 'asc') ? true : undefined} value='asc'>Name: A-Z</option>
```
Comment (sherakama) [inline src/eventshub/src/component/SortBy/SortBy.tsx]: I punted and added aria-selected here. Ugly yes, and I have to check in to see if CJWest thinks this is valid or if I should go flog myself for trying something so dastardly.

File: src/eventshub/src/index.css [hunk 9]
```diff
@@ -178,6 +183,10 @@
   @apply er-text-black er-border-b-3 er-border-black;
 }
 
+#su-eventshub .aa-Panel {
+  @apply er-z-50 er-max-w-[100%] er-min-w-[100%] er-w-[100%];
```
Comment (sherakama) [inline src/eventshub/src/index.css:187]: Belt and suspenders and duct-tape for good measure. 😊

File: src/eventshub/src/component/SortBy/SortBy.tsx (hunk 8 above)
Comment (cjwest) [reply src/eventshub/src/component/SortBy/SortBy.tsx]: Haha! Please don't flog yourself, @sherakama!
I don't know this code enough to see what dangers lurk. Let me know where I can look at this when available. 🍀

**Comment by sherakama** [comment]
FYI: Got to the pager A11y in https://github.com/SU-SWS/eirh-subtheme/pull/65

File: src/eventshub/src/component/SortBy/SortBy.tsx (hunk 8 above)
Comment (sherakama) [reply src/eventshub/src/component/SortBy/SortBy.tsx]: Went with it! Thanks.

**Review by rebeccahongsf — APPROVED** [review]
Autocomplete is functioning as expected.
I added and updated some of the aria-labels. We should be GTG but I'll put all the a11y fixes onto the preview site so that @cjwest can review 😃

**Pull request merged**